client/server/daemon.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/wiretrustee/wiretrustee/client/internal"
+	"github.com/wiretrustee/wiretrustee/client/proto"
+)
+
+// Server for service control.
+type Server struct {
+	managementURL string
+	configPath    string
+	stopCh        chan int
+	cleanupCh     chan<- struct{}
+
+	config *internal.Config
+	proto.UnimplementedDaemonServiceServer
+}
+
+// New server instance constructor.
+func New(managementURL, configPath string, stopCh chan int, cleanupCh chan<- struct{}) *Server {
+	return &Server{
+		managementURL: managementURL,
+		configPath:    configPath,
+		stopCh:        stopCh,
+		cleanupCh:     cleanupCh,
+	}
+}
+
+// Login uses setup key to prepare configuration for the daemon.
+func (s *Server) Login(ctx context.Context, msg *proto.LoginRequest) (*proto.LoginResponse, error) {
+	config, err := internal.GetConfig(s.managementURL, s.configPath, msg.PresharedKey)
+	if err != nil {
+		return nil, err
+	}
+	s.config = config
+
+	// login operation uses backoff scheme to connect to management API
```

Comment:
so it will always return success to the login command?
not sure if it is good.
Maybe we should move it to the cmd and retry if failed to log in?

Reply:
also if we trigger Login first, 
it will return immediately
then up command will run and could fail because the server didn't register the peer yet.
the UP command backoff helps but still...